app/models/ride.rb

```diff
@@ -1,3 +1,12 @@
 class Ride < ApplicationRecord
   belongs_to :amusement_park
+  has_many :ride_mechanics
+  has_many :mechanics, through: :ride_mechanics
+
+  def self.ordered_by_mechanic
+    joins(:mechanics)
+    .select("rides.*, avg(mechanics.years_experience) AS average_experience")
+    .group("rides.id")
+    .order("average_experience desc")
+  end
```

Comment:
This is a great query, nicely done! Making it a class method on the `Ride` model was a nice touch.